docs/source/guides/quickstart.md

```diff
@@ -1,48 +1,149 @@
 ```
 
-### Configure a Persistent Ursula
+## Alice: Grant Access to a Secret
 
-```bash
-$ nucypher ursula init --federated-only
+```python
+from nucypher.characters.lawful import Alice, Bob, Ursula
+
+# Application Side-Channel
+# --------------------------
+# bob_encrypting_key = <Side Channel>
+# bob_verifying_key = <Side Channel>
+
+ursula = Ursula.from_seed_and_stake_info(seed_uri='https://0.0.0.0:9151')
+alice = Alice(checksum_address='0xdeadbeef', 
+              provider_uri=<PROVIDER URI>,
+              known_nodes=[ursula])
+bob = Bob.from_public_keys(verifying_key=bob_verifying_key, encrypting_key=bob_encrypting_key)
 ```
 
-### Run a Persistent Ursula
+```python
+from datetime import datetime, timedelta
 
-```bash
-$ nucypher ursula run --network <NETWORK_DOMAIN> --teacher <SEEDNODE_URI> --federated-only
+#
+# Grant
+#
+
+expiration = datetime.now() + timedelta(days=5)  # Five days from now.
+policy = alice.grant(bob,
+                     label=b'my-secret-stuff',
+                     m=2, n=3,
+                     expiration=expiration)
+
+policy_encrypting_key = policy.public_key
 ```
 
-Replace `<NETWORK_DOMAIN>` with the network domain and `<SEEDNODE_URI>` with the URI of a node running on that network
-domain you want to connect to (for example `0.0.0.0:9151` or `0xdeadbeef@0.0.0.0:9151`).
+Note that Alice can get the public key even before creating the policy.
+From this moment on, any Data Source (Enrico) that knows the public key
+can encrypt data originally intended for Alice, but can be shared with
+any Bob that Alice grants access.
 
-If you're connecting to the `devnet`, you should use `--network devnet --teacher 18.222.119.242:9151`.
+`policy_pubkey = alice.get_policy_encrypting_key_from_label(label)`
 
-### Run a Geth-Connected Development Ursula
 
-Run a local geth node in development mode:
+## Enrico: Encrypt a Secret
 
-```bash
-$ geth --dev
+```python
+from nucypher.characters.lawful import Enrico
+
+# Application Side-Channel
+# --------------------------
+# policy_encrypting_key = <Side Channel>
+
+enrico = Enrico(policy_encrypting_key=policy_encrypting_key)
+ciphertext = enrico.encrypt_message(message=b'Peace at dawn.')
 ```
 
-Run a local development Ursula connected to the geth node
 
-```bash
-$ nucypher ursula run --dev --provider ipc:///tmp/geth.ipc --checksum-address <GETH_DEV_ADDRESS>
+## Bob: Decrypt a Secret
+
+```python
+from nucypher.characters.lawful import Alice, Bob, Enrico, Ursula
+
+
+# Application Side-Channel
+# --------------------------
+# label = <Side Channel>
+# ciphertext = <Side Channel>
+# policy_encrypting_key = <Side Channel>
+# alice_verifying_key = <Side Channel>
+
+# Everyone!
+ursula = Ursula.from_seed_and_stake_info(seed_uri='https://0.0.0.0:9151')
+alice = Alice.from_public_keys(verifying_key=alice_verifying_key)
+bob = Bob(checksum_address='0xdeadbeef',
+          provider_uri=<PROVIDER URI>,
+          known_nodes=[ursula])
```

Comment:
Bob also needs to convert public keys into transportable form:
```python
from nucypher.crypto.powers import DecryptingPower, SigningPower
encryption_key_hex = bob.public_keys(DecryptingPower).hex()
verifying_key_hex = bob.public_keys(SigningPower).hex()
```

Reply:
Of course, Bob needs to be with a learning loop and registry, just like Alice:
```python
bob = Bob(checksum_address='0x900A10C9De490487d0516aF58Eb17ea3106B6041', known_nodes=[ursula], network_middleware=RestMiddleware(), registry=registry)
bob.start_learning_loop()
```
Then `len(bob.known_nodes)` will show something more than 1